### DISCUSSION TITLE
6.1 kernel & kvm

### DISCUSSION THREAD

**Discussion by mwoodpatrick** [microsoft/WSL#10950] [Q&A]
I'm following the article:

[How-to use the Microsoft Linux kernel v6 on WSL2 | Microsoft Learn](https://learn.microsoft.com/en-us/community/content/wsl-user-msft-kernel-v6)

The branch mentioned in the article v6.1.x no longer exists so I'm using the linux-msft-wsl-6.1.y branch and during the build, it asks a few questions including:

Mitigate Straight-Line-Speculation (SLS)

What is the recommended answer for this?

During the build I hit the issue:
```

BTF: .tmp_vmlinux.btf: pahole (pahole) is not available
Failed to generate BTF for vmlinux
Try to disable CONFIG_DEBUG_INFO_BTF
make[1]: *** [scripts/Makefile.vmlinux:34: vmlinux] Error 1
make: *** [Makefile:1255: vmlinux] Error 2
```
See bug: https://github.com/microsoft/WSL/issues/7667. The same fix appears to work:

sudo apt install dwarves

I'm also looking at the articles:

[Build An Accelerated KVM Guest Custom Kernel for WSL 2 - 2022 Edition (boxofcables.dev)](https://boxofcables.dev/kvm-optimized-custom-kernel-wsl2-2022/)

[Compiling your own WSL2 Kernel for Cilium (harthoover.com)](https://harthoover.com/compiling-your-own-wsl2-kernel/)

It would be useful if anyone has any suggestions on how to apply these to the 6.1 kernel.

Any pointers on best way to use this kernel and KVM module with libvirt and virt-manager would be useful

**Comment by pbk20191** [comment]
Hi, I use wsl 6.1 and I suppose it already has kvm feature.
I can not say if it actually works, because I'm not sure what I can do to confirm it is working.
But I think you can check it by yourself. 

My hardware support Virtualization
```
$ lscpu | grep Virtualization
Virtualization:                  AMD-V
Virtualization type:             full
$ kvm-ok
INFO: /dev/kvm exists
KVM acceleration can be used
```

wsl 6.1 has multiple kvm patch [commit](https://github.com/microsoft/WSL2-Linux-Kernel/commit/579da60d2e21679c6f80d32f68724c46b5d5e468)

and this is the piece of 6.1 wsl kernel config
```make
CONFIG_HAVE_KVM=y
CONFIG_HAVE_KVM_PFNCACHE=y
CONFIG_HAVE_KVM_IRQCHIP=y
CONFIG_HAVE_KVM_IRQFD=y
CONFIG_HAVE_KVM_IRQ_ROUTING=y
CONFIG_HAVE_KVM_DIRTY_RING=y
CONFIG_HAVE_KVM_DIRTY_RING_TSO=y
CONFIG_HAVE_KVM_DIRTY_RING_ACQ_REL=y
CONFIG_HAVE_KVM_EVENTFD=y
CONFIG_KVM_MMIO=y
CONFIG_KVM_ASYNC_PF=y
CONFIG_HAVE_KVM_MSI=y
CONFIG_HAVE_KVM_CPU_RELAX_INTERCEPT=y
CONFIG_KVM_VFIO=y
CONFIG_KVM_GENERIC_DIRTYLOG_READ_PROTECT=y
CONFIG_KVM_COMPAT=y
CONFIG_HAVE_KVM_IRQ_BYPASS=y
CONFIG_HAVE_KVM_NO_POLL=y
CONFIG_KVM_XFER_TO_GUEST_WORK=y
CONFIG_VIRTUALIZATION=y
CONFIG_KVM=y
CONFIG_KVM_WERROR=y
CONFIG_KVM_INTEL=y
CONFIG_KVM_AMD=y
# CONFIG_KVM_XEN is not set
```


I can see my kvm moudles
```
$ /sys/module/kvm
kvm/       kvm_amd/   kvm_intel/ 
```
